lib/f_service/base.rb

```diff
@@ -25,13 +20,35 @@ class Base
       #
       # @note this method shouldn't be overridden in the subclasses
       # @return [Result::Success, Result::Failure]
-      def self.call(#{all_args})
-        result = new(#{all_args}).run
+      def call(*args)
+        result = new(*args).run
         raise(FService::Error, 'Services must return a Result') unless result.is_a? Result::Base
 
         result
       end
-    RUBY
+
+      # Allows running a service without explicit giving params.
+      # This is useful when chaining services or mapping inputs to be processed.
+      #
+      # @example
+      #   # Assuming all classes here subclass FService::Base:
+      #
+      #   User::Create
+      #     .then(&User::Login)
+      #     .then(&SendWelcomeEmail)
+      #
+      #   # Mapping inputs:
+      #
+      #   [{ n:1 }, { n: 2 }].map(&DoubleNumber).map(&:value)
+      #   # => [2, 4]
+      #
+      # @return [Proc]
+      def to_proc
+        proc { |args| call(**args) }
+      end
+
+      ruby2_keywords :call if respond_to?(:ruby2_keywords, true)
```

Comment:
Can you move this right after the `def call` itself?

Reply:
Done.